vividus-plugin-browserstack/src/main/java/org/vividus/browserstack/session/BrowserStackSessionStatusManager.java

```
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.browserstack.session;
+
+import com.browserstack.client.exception.BrowserStackException;
+
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.vividus.browserstack.BrowserStackAutomateClient;
+import org.vividus.selenium.IWebDriverProvider;
+import org.vividus.testcontext.TestContext;
+import org.vividus.ui.session.AbstractSessionStatusManager;
+
+public class BrowserStackSessionStatusManager extends AbstractSessionStatusManager
+{
+    private final BrowserStackAutomateClient browserStackAutomateClient;
+
+    public BrowserStackSessionStatusManager(IWebDriverProvider webDriverProvider, TestContext testContext,
+            BrowserStackAutomateClient browserStackAutomateClient)
+    {
+        super(webDriverProvider, testContext);
+        this.browserStackAutomateClient = browserStackAutomateClient;
+    }
+
+    @Override
+    protected void updateSessionStatus(String status)
+    {
+        try
+        {
+            String sessionId = getWebDriverProvider().getUnwrapped(RemoteWebDriver.class).getSessionId().toString();
+            browserStackAutomateClient.updateSessionStatus(sessionId, status);
+        }
+        catch (BrowserStackException e)
+        {
+            throw new IllegalStateException(e);
```

it should swallow exception quietly